api/tests/routes/backoffice/accounts_test.py

```diff
@@ -0,0 +1,190 @@
+from unittest import mock
+
+from flask import url_for
+import pytest
+
+from pcapi.core.permissions.factories import RoleFactory
+from pcapi.core.permissions.models import Permission
+from pcapi.core.permissions.models import Permissions
+from pcapi.core.users import factories as user_factories
+from pcapi.repository import repository
+
+
+pytestmark = pytest.mark.usefixtures("db_session")
+
+
+def create_bunch_of_accounts():
+    underage = user_factories.UnderageBeneficiaryFactory(
+        firstName="Gédéon", lastName="Groidanlabénoir", email="gg@example.com", phoneNumber="0123456789"
+    )
+    grant_18 = user_factories.BeneficiaryGrant18Factory(
+        firstName="Abdel Yves Akhim", lastName="Flaille", email="ayaf@example.com", phoneNumber="0516273849"
+    )
+    pro = user_factories.ProFactory(
+        firstName="Gérard", lastName="Mentor", email="gm@example.com", phoneNumber="0246813579"
+    )
+    random = user_factories.UserFactory(
+        firstName="Anne", lastName="Algézic", email="aa@example.com", phoneNumber="0606060606"
+    )
+
+    return underage, grant_18, pro, random
+
+
+def create_search_role():
+    permission = Permission.query.filter_by(name=Permissions.SEARCH_PUBLIC.value).first()
+    role = RoleFactory(name="public search")
+    role.permissions.append(permission)
+    repository.save(role)
+
+    return role
+
+
+class PublicAccountSearchTest:
```

Comment:
Je n'ai pas vu de cas de test où on cherchais sur une partie de nom ou prénom, mais c'est peut-être pas un usecase métier ?

Reply:
j'ai pas testé ce cas effectivement, je peux rajouter ça

Reply:
ceci dit, je suis convaincu qu'il va vite nous falloir une recherche un peu plus maline que celle-ci:
si on cherche un nom avec un prénom, ça va nous retourner tous les gens qui ont ce nom + tous les gens qui ont ce prénom.

perso ça me parait foireux, mais je me suis aligné sur ce que propose FlaskAdmin pour le moment car un système de ranking avec éventuellement de la recherche plein-texte aurait été trop long à faire pour une premièere itération.

bref, vu qu'on risque fort d'avoir un changement de mécanisme de recherche à faire prochainement, est-ce que ça vaut vraiment le coup de blinder de tests qui vont vite être obsolète? (d'autant que pour le coup, je suis très confiant quant au fait que ça va fonctionner aussi avec des noms/prénoms incomplets)